contracts/roles/RevocationRegistryOnChain.sol

```diff
@@ -0,0 +1,97 @@
+        bytes32 revokerRoleDigest; //namehash
+    }
+
+    mapping(bytes32 => RevokedClaim) revokedClaims;
+    address private didRegistry;
+    address private ensRegistry;
+    address private claimManager;
+    
+    event Revoked(address revoker, bytes32 claimDigest);
+
+    constructor(address _didRegistry, address _ensRegistry, address _claimManager){
+        didRegistry = _didRegistry;
+        ensRegistry = _ensRegistry;
+        claimManager = _claimManager;
+    }
+
+    function revokeClaim(bytes32 claimDigest, bytes32 role, address revoker, bytes32 revokerRoleDigest) public {
+        require(revokedClaims[claimDigest].revokedTimestamp == 0, "The claim is already revoked");
+        verifyRevoker(revoker, role, revokerRoleDigest);
+        revokedClaims[claimDigest].revoker = revoker;
+        revokedClaims[claimDigest].revokedTimestamp = block.number;
+        revokedClaims[claimDigest].revokerRoleDigest = revokerRoleDigest;
+        emit Revoked(revoker, claimDigest);
+    }
+
+    function revokeClaimInList(bytes32 [] memory claimList,bytes32 role, address revoker, bytes32 revokerRoleDigest) public{
+        verifyRevoker(revoker, role, revokerRoleDigest);
+        for(uint i=0; i<claimList.length ; i++) {
+            require(revokedClaims[claimList[i]].revokedTimestamp == 0, "The claim is already revoked");
+            revokedClaims[claimList[i]].revoker = revoker;
+            revokedClaims[claimList[i]].revokedTimestamp = block.number;
+            revokedClaims[claimList[i]].revokerRoleDigest = revokerRoleDigest;
+            emit Revoked(revoker, claimList[i]);
+        }
+    }
+
+    function verifyRevoker(address revoker, bytes32 role, bytes32 revokerRoleDigest) internal view {
+        address resolver = ENSRegistry(ensRegistry).resolver(role);
+        (address[] memory dids, bytes32 revoker_role) = RevokersResolver(resolver).revokers(role);
+        if (dids.length > 0) {
+          EthereumDIDRegistry registry = EthereumDIDRegistry(didRegistry);
+            for (uint i = 0; i < dids.length; i++) {
+                if (dids[i] == revoker || registry.validDelegate(dids[i], ASSERTION_DELEGATE_TYPE, revoker)) {
+                    return;
+                }
+            }
+            revert("Revocation Registry: Revoker is not listed in role revokers list");
+        } else if (revoker_role != "") {
+            ClaimManager cm = ClaimManager(claimManager);
+            bool hasRole = cm.hasRole(revoker, revoker_role, 0);
+            if (hasRole) {
+                bool roleStatus = isRevoked(revokerRoleDigest);
+                if (!roleStatus) {
+                    return;
+                } else {
+                    revert("Revocation Registry: Revoker's role has been revoked");
+                }
+            }
+            revert("Revocation Registry: Revoker does not have required role");
+        } else {
+            revert("Revocation Registry: Role revokers are not specified");
+        }
+    }
```

Comment:
Conceptually, I find it wired to `revert` a view function.
What do you think of this pattern?
```suggestion
    function verifyRevoker(address revoker, bytes32 role, bytes32 revokerRoleDigest) internal view {
        address resolver = ENSRegistry(ensRegistry).resolver(role);
        (address[] memory dids, bytes32 revoker_role) = RevokersResolver(resolver).revokers(role);
        if (dids.length > 0) {
          EthereumDIDRegistry registry = EthereumDIDRegistry(didRegistry);
            for (uint i = 0; i < dids.length; i++) {
                bool revokerListed = (dids[i] == revoker || registry.validDelegate(dids[i], ASSERTION_DELEGATE_TYPE, revoker));
                require(revokerListed, "Revocation Registry: Revoker is not listed in role revokers list");
            }
        } else {
            require(revoker_role != "", "Revocation Registry: Role revokers are not specified");
            ClaimManager cm = ClaimManager(claimManager);
            bool hasRole = cm.hasRole(revoker, revoker_role, 0);
            require(hasRole, "Revocation Registry: Revoker does not have required role");
            bool isNotRevoked = !isRevoked(revokerRoleDigest);
            require(isNotRevoked, "Revocation Registry: Revoker's role has been revoked");
        }
    }
```

Reply:
Below are my thoughts on using revert() irrespective of the function type:
1. The require() methods compile to the REVERT OPCODE at the end.
2. To validate the input and constraints it's good to use require(). For the case you want to validate data from another contract which does involve business logic or validations, it is recommended to use revert().
3. Also, in the ClaimManager contract, we are using revert() in verifyIssuer as well. This makes it consistent with the previous implemented contract.